transform/variant/maf_transform.py

```diff
@@ -0,0 +1,226 @@
+import logging
+import csv
+import gzip
+import sys
+
+from bmeg.models.vertex_models import Biosample, Callset
+from bmeg.models.edge_models import AlleleCall, CallsetFor
+from bmeg.models.emitter import Emitter
+
+import allele_harvester
+
+from concurrent.futures import ThreadPoolExecutor
+from concurrent.futures import as_completed
+
+
+def get_value(d, keys, default):
+    """ utility get value from list"""
+    if isinstance(keys, list):
+        for k in keys:
+            if k in d:
+                return d[k]
+    elif keys in d:
+        return d[keys]
+    return default
+
+
+def _multithreading(func, lines, max_workers=5):
+    """
+    Create a thread pool and create alleles
+    """
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        futures = (executor.submit(func, line) for line in lines)
+        for future in as_completed(futures):
+            yield future.result()
+
+
+def _read_maf(mafpath, gz):
+    """ generator for each line in maf """
+    if gz or 'gz' in mafpath:
+        inhandle = gzip.open(mafpath, mode='rt')
+    else:
+        inhandle = open(mafpath)
+    reader = csv.DictReader(inhandle, delimiter="\t")
+    for line in reader:
+        yield line
+    inhandle.close()
+
+
+class MafConverter():
+
+    def convert(self, emit, mafpath, source, genome='GRCh37',
+                method='variant', gz=False, centerCol='Center'):
+        """
+        emit -  a way to write output
+        mafpath - a file to read
+        source - special handling if 'tcga'
+        genome - reference_genome e.g. GRCh37
+        method - call's method
+        gz - is mafpath a gz
+        """
+        # center = 2
+        # ncbi_build = 3
+        chromosome = "Chromosome"  # 4
+        start = ["Start_Position", "Start_position"]  # 5
+        end = ["End_Position", "End_position"]  # 6
+        # strand = 7
+        variant_type = "Variant_Type"  # 9
+        reference_allele = "Reference_Allele"  # 10
+        # tumor_allele1 = "Tumor_Seq_Allele1"  # 11
+        tumor_allele2 = "Tumor_Seq_Allele2"  # 12
+        # annotation_transcript = "Annotation_Transcript"  # 14
+
+        tumor_sample_barcode = "Tumor_Sample_Barcode"  # 15
+        normal_sample_barcode = "Matched_Norm_Sample_Barcode"  # 16
+
+        # normal_allele1 = 17
+        # normal_allele2 = 18
+        # verification_status = 23
+        # validation_status = 24
+        # mutation_status = 25
+        # sequencing_phase = 26
+        # sequence_source = 27
+        # bam_file = 30
+
+        # sequencer = 31
+        # genome_change = 32
+
+        # # Information indices for VariantCallEffect and Gene
+        # entrez_gene_id = 1
+        # variant_classification = 8
+
+        feature = "Feature"  # 48
+        feature_type = "Feature_type"  # 49
+        dbSNP_RS = "dbSNP_RS"  # 13
+
+        logging.info('converting maf: ' + mafpath)
+
+        sample_ids = {}
+
+        def _allele_call_maker(allele, callset, line=None):
```

Comment:
These can be moved to the root level, there's no need to make everything private and deeply nested.

Reply:
the function when defined in scope has access to locals().  since this one is parsing the `line` it needs access to constants